src/pages/FrontendMentor/TipCalculatorApp/index.tsx

```diff
@@ -0,0 +1,207 @@
+    setNewPeopleAmount(e.target.valueAsNumber)
+  }
+
+  const handleSetCustomAmount = (e: ChangeEvent<HTMLInputElement>) => {
+    setCustomAmount(e.target.valueAsNumber)
+  }
+
+  const errorString = "Can't be zero"
+
+  return (
+    <MainContainer>
+      <Header>
+        <HeaderWords>
+          S P L I<br />T T E R
+        </HeaderWords>
+      </Header>
+      <MiddleContainer>
+        <Calculator>
+          <InteractionContainer>
+            <ItemContainer>
+              <ItemHeader>Bill</ItemHeader>
+              <InputContainer>
+                <InputIcon src={DollarIcon} alt="dollar sign"></InputIcon>
+                <form>
+                  <InputField
+                    value={newBillAmount}
+                    type="number"
+                    placeholder={billAmount.toString()}
+                    onChange={(e) => handleNewBillAmount(e)}
+                  />
+                </form>
+              </InputContainer>
+            </ItemContainer>
+            <ItemContainer>
+              <ItemHeader>Select Tip %</ItemHeader>
+              <TipGridBox>
+                <TipGridItem
+                  onClick={() => {
+                    dispatch(setBillAmount(newBillAmount))
+                    dispatch(setPeopleAmount(newPeopleAmount))
+                    dispatch(setTipAmount(5))
+                    dispatch(setTotalAmount())
+                  }}
+                >
+                  5%
+                </TipGridItem>
+                <TipGridItem
+                  onClick={() => {
+                    dispatch(setBillAmount(newBillAmount))
+                    dispatch(setPeopleAmount(newPeopleAmount))
+                    dispatch(setTipAmount(10))
+                    dispatch(setTotalAmount())
+                  }}
+                >
+                  10%
+                </TipGridItem>
+                <TipGridItem
+                  onClick={() => {
+                    dispatch(setBillAmount(newBillAmount))
+                    dispatch(setPeopleAmount(newPeopleAmount))
+                    dispatch(setTipAmount(15))
+                    dispatch(setTotalAmount())
+                  }}
+                >
+                  15%
+                </TipGridItem>
+                <TipGridItem
+                  onClick={() => {
+                    dispatch(setBillAmount(newBillAmount))
+                    dispatch(setPeopleAmount(newPeopleAmount))
+                    dispatch(setTipAmount(25))
+                    dispatch(setTotalAmount())
+                  }}
+                >
+                  25%
+                </TipGridItem>
+                <TipGridItem
+                  onClick={() => {
+                    dispatch(setBillAmount(newBillAmount))
+                    dispatch(setPeopleAmount(newPeopleAmount))
+                    dispatch(setTipAmount(50))
+                  }}
+                >
+                  50%
+                </TipGridItem>
```

Comment:
these look q similar, can consider making an object and map them to reduce duplicated code; but works too

Reply:
Will try it next time! 👍🏼